lib/rucio/common/logging.py

```diff
@@ -344,15 +363,15 @@ def __init__(
         self.output_json = output_json
         super().__init__(fmt=fmt, style='%', **_kwargs)
 
-    def format(self, record):
-        json_record = dict(itertools.chain.from_iterable(f.format(record) for f in self._desired_data_sources))
+    def format(self, record: "LogRecord") -> str:
+        json_record = dict(itertools.chain.from_iterable(f.format(record) for f in self._desired_data_sources))  # type: ignore
```

Comment:
The `type: ignore` here is due to the fact that `LogDataSource.format` returns `None` if there is no `self._formatter`, and the static type checker is not able to guarantee that `f` has a `_formatter` specified in this case. 

This logic should likely be changed so that it only runs `LogDataSource.format` if `_formatter` exists, and otherwise returns `None`, but I think it's best to just`type: ignore` for now and address this later.